sketches/src/main/java/org/apache/apex/malhar/sketches/QuantilesEstimator.java

```diff
@@ -0,0 +1,188 @@
+ * <br>+
+ */
+@OperatorAnnotation(partitionable = false)
+public class QuantilesEstimator extends BaseOperator
+{
+  /**
+   * Constructor that allows non-default initialization of the quantile sketch object
+   *
+   * @param k:    Parameter that determines accuracy and memory usage of quantile sketch. See QuantilesSketch
+   *              documentation for details
+   * @param seed: The quantile sketch algorithm is inherently random. Set seed to 0 for reproducibility in testing, but
+   *              do not set otherwise.
+   */
+  public QuantilesEstimator(int k, short seed)
+  {
+    quantilesSketch = QuantilesSketch.builder().setSeed(seed).build(k);
+  }
+
+  private transient QuantilesSketch quantilesSketch = QuantilesSketch.builder().build();
+
+  /**
+   * This field determines the specific quantiles to be calculated. For a stream of numbers, the quantile at a value
+   * 0 <= p <= 1 is the number x such that a fraction p of the numbers in the sorted stream are less than x. E.g., the
+   * quantile at p = 0.5 is the median (half the numbers in the stream are less than the median).
+   * The default is set to compute the standard quartiles (4-quantiles).
+   */
+  private double[] fractions = {0.0, 0.25, 0.50, 0.75, 1.00};
+  /**
+   * This field determines the intervals on which the probability mass function is computed.
+   */
+  private double[] pmfIntervals = {};
+
+  /**
+   * This operator computes three different quantities which are output on separate output ports. If not using any of
+   * these quantities, these variables can be set to avoid unnecessary computation.
+   */
+  private boolean computeCdf = true;
+  private boolean computeQuantiles = true;
+  private boolean computePmf = true;
+
+  public boolean isComputeCdf()
+  {
+    return computeCdf;
+  }
+
+  public void setComputeCdf(boolean computeCdf)
+  {
+    this.computeCdf = computeCdf;
+  }
+
+  public boolean isComputeQuantiles()
+  {
+    return computeQuantiles;
+  }
+
+  public void setComputeQuantiles(boolean computeQuantiles)
+  {
+    this.computeQuantiles = computeQuantiles;
+  }
+
+  public boolean isComputePmf()
+  {
+    return computePmf;
+  }
+
+  public void setComputePmf(boolean computePmf)
+  {
+    this.computePmf = computePmf;
+  }
+
+  public int getK()
+  {
+    return quantilesSketch.getK();
+  }
+
+  public double[] getFractions()
+  {
+    return fractions;
+  }
+
+  public void setFractions(double[] fractions)
+  {
+    this.fractions = fractions;
+  }
+
+  public double[] getPmfIntervals()
+  {
+    return pmfIntervals;
+  }
+
+  public void setPmfIntervals(double[] pmfIntervals)
+  {
+    this.pmfIntervals = pmfIntervals;
+  }
+
+  /**
+   * Returns total number of elements seen in stream so far.
+   */
+  public long streamLength()
+  {
+    return quantilesSketch.getN();
+  }
+
+  public final transient DefaultInputPort<Double> data = new DefaultInputPort<Double>()
```

Comment:
Please reorder all declarations in the following order: 
data members, ports, constructors, all other methods, getters and setters.